### ISSUE TITLE
New Feature:  More stats

### ISSUE THREAD

**Issue by iinsertNameHere** [iinsertNameHere/catnip#7]
More stats:
- PKG manager
- PKG count
- Memory
- CPU
- GPU
- Device
- Disk space

**Comment by yu-kiat** [comment]
Ways to get stats:
- PKG Manager: I don't think there's a way to do it unless we do it manually, with perhaps `detectos()` from `std/distros` and manually assigning the package manager based on the output.
- PKG Count: After doing the above, we could use the respective command to get the total number of packages (for example `pacman -Q`).
- Terminal: Using the `$TERM` environment variable.

**Comment by iinsertNameHere** [comment]
I think it is best to don't rely on env vars as much as possible so I would just get the parent of parent of the current process if that makes sense. ; )

**For example:**
`kitty -> bash -> catnip`
**Kitty** would be the parent of the parent of the current (catnip) process.

**Comment by JustBuddy** [comment]
At least for the debian package systems, installed packages can be called with:
`dpkg --list | wc --lines`
Any system using dpkg should be able to do this.

![image](https://github.com/iinsertNameHere/catnip/assets/8122264/c9fbc7cd-d67a-4491-8b26-f2912ae24282)

**Comment by iinsertNameHere** [comment]
@JustBuddy Yes, that is true. It is not really a fetch tool if you have to define the information..

**Comment by JustBuddy** [comment]
So the workflow could be something like:

1. Probe for used package manager. (Support the most common ones)
- If no supported package manager could be detected, omit the stats from the fetch OR show as unsupported.
2. Fetch the package count from the package manager discovered in 1.
3. Display in fetch.

I am not sure if we need to display the package manager itself in a fetch, that is usually non-dynamic, as in: It is unlikely that this value will change in an instance. If we do, it could be displayed in a single line like
| 📦 packages | apt - 1337 installed

**Comment by iinsertNameHere** [comment]
@JustBuddy I also thought about something like that. But if the manager should be displayed I think your idea to save a row would be best.
